src/System.Threading.Channels/tests/Stress.cs

```diff
@@ -0,0 +1,423 @@
+            yield return new object[] { Channel.CreateBounded<int>(BoundedDropWriteSingleReaderOptions), BoundedDropWriteSingleReaderOptions };
+            yield return new object[] { Channel.CreateBounded<int>(BoundedDropWriteSingleWriterOptions), BoundedDropWriteSingleWriterOptions };
+            yield return new object[] { Channel.CreateBounded<int>(BoundedDropWriteSingleReaderWriterOptions), BoundedDropWriteSingleReaderWriterOptions };
+
+            //
+            // Bounded Channel ( AllowSynchronousContinuations,  BoundedChannelFullMode.DropWrite)
+            //
+            yield return new object[] { Channel.CreateBounded<int>(BoundedSynchronousDropWriteOptions), BoundedSynchronousDropWriteOptions };
+            yield return new object[] { Channel.CreateBounded<int>(BoundedSynchronousDropWriteSingleReaderOptions), BoundedSynchronousDropWriteSingleReaderOptions };
+            yield return new object[] { Channel.CreateBounded<int>(BoundedSynchronousDropWriteSingleWriterOptions), BoundedSynchronousDropWriteSingleWriterOptions };
+            yield return new object[] { Channel.CreateBounded<int>(BoundedSynchronousDropWriteSingleReaderWriterOptions), BoundedSynchronousDropWriteSingleReaderWriterOptions };
+
+            //
+            // Unbuffered Channel
+            //
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedDefaultOptions), UnbufferedDefaultOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSingleReaderOptions), UnbufferedSingleReaderOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSingleReaderWriterOptions), UnbufferedSingleReaderWriterOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSingleWriterOptions), UnbufferedSingleWriterOptions };
+
+            //
+            // Unbuffered Channel (AllowSynchronousContinuations)
+            //
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSynchronousOptions), UnbufferedSynchronousOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSynchronousSingleReaderOptions), UnbufferedSynchronousSingleReaderOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSynchronousSingleReaderWriterOptions), UnbufferedSynchronousSingleReaderWriterOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSynchronousSingleWriterOptions), UnbufferedSynchronousSingleWriterOptions };
+            yield return new object[] { Channel.CreateUnbuffered<int>(UnbufferedSynchronousOptions), UnbufferedSynchronousOptions };
+        }
+
+        const int MAX_NUMBER_TO_WRITE_TO_CHANNEL = 1_000_000;
+        private static readonly int MaxTaskCounts = Math.Max(2, Environment.ProcessorCount);
+
+        [OuterLoop]
```

Comment:
How long does this test take to run?  If it takes a long time, even in outerloop that could be problematic.  And if it doesn't take a long time, what is it stressing?

Reply:
No, it doesn't take that long. on my machine, it takes less than 2 minutes. it should be faster after we remove the unbuffered channel too. so I think keeping it in the outerloop is ok.

Reply:
> it takes less than 2 minutes

I think that's way too long, even for an Outerloop test.

@danmosemsft, @karelz, what's the recommendation for stress tests in corefx?

Reply:
I just measured it (for the 1,000,000 iterations) and it took 1 minute.